render_silhouette_regmarks.py

```diff
@@ -0,0 +1,132 @@
+from inkex import EffectExtension, Boolean, Rectangle, Line, PathElement, Layer, Group, TextElement, Polygon, Transform
+from gettext import gettext
+
+REGMARK_LAYERNAME = 'Regmarks'
+REGMARK_LAYER_ID = 'regmark'
+REGMARK_TOP_LEFT_ID = 'regmark-tl'
+REGMARK_TOP_RIGHT_ID = 'regmark-tr'
+REGMARK_BOTTOM_LEFT_ID = 'regmark-bl'
+REGMARK_SAFE_AREA_ID = 'regmark-safe-area'
+REGMARK_NOTES_ID = 'regmark-notes'
+
+REG_SQUARE_MM = 5
+REG_LINE_MM = 20
+REG_SAFE_AREA_MM = 20
+
+# https://www.reddit.com/r/silhouettecutters/comments/wcdnzy/the_key_to_print_and_cut_success_an_extensive/
+# > The registration mark thickness is actually very important. For some reason, 0.3 mm marks work perfectly. 
+# > The thicker you get, the less accurate registration will be. ~~~ galaxyman47
+REG_MARK_LINE_WIDTH_MM = 0.3
+
+REG_MARK_INFO_FONT_SIZE_PT = 2.5
+
+class InsertRegmark(EffectExtension):
+	def __init__(self):
+		EffectExtension.__init__(self)
+
+	def add_arguments(self, pars):
+		# Parse arguments
+		pars.add_argument("-X", "--reg-x", "--regwidth",  type = float, dest = "regwidth",   default = 180.0, help="X mark distance [mm]")
+		pars.add_argument("-Y", "--reg-y", "--reglength", type = float, dest = "reglength",  default = 230.0, help="Y mark distance [mm]")
+		pars.add_argument("--rego-x",  "--regoriginx",    type = float, dest = "regoriginx", default = 15.0,  help="X mark origin from left [mm]")
+		pars.add_argument("--rego-y", "--regoriginy",     type = float, dest = "regoriginy", default = 20.0,  help="X mark origin from top [mm]")
+		pars.add_argument("--verbose", dest = "verbose",  type = Boolean, default = False, help="enable log messages")
+
+	#SVG SVGd from (x,y) dimentional points
+	def points_to_svgd(self, p, capped=False):
+		f = p[0]
+		p = p[1:]
+		svgd = "M{:.5f},{:.5f}".format(f[0], f[1])
+		for x in p:
+			svgd += " L{:.5f},{:.5f}".format(x[0], x[1])
+		if capped:
+			svgd += "z"
+		return svgd
+
+	def effect(self):
+		reg_origin_X = self.options.regoriginx
+		reg_origin_Y = self.options.regoriginy
+		reg_width = self.options.regwidth if self.options.regwidth else int(self.svg.get("width").rstrip("mm")) - reg_origin_X*2
+		reg_length = self.options.reglength if self.options.reglength else int(self.svg.get("height").rstrip("mm")) - reg_origin_Y*2
+
+		if self.options.verbose == True:
+			self.msg(gettext("[INFO]: page width ")+str(self.svg.get("width").rstrip("mm")))
+			self.msg(gettext("[INFO]: page height ")+str(self.svg.get("height").rstrip("mm")))
+			self.msg(gettext("[INFO]: regmark from document left ")+str(reg_origin_X))
+			self.msg(gettext("[INFO]: regmark from document top ")+str(reg_origin_Y))
+			self.msg(gettext("[INFO]: regmark to regmark spacing X ")+str(reg_width))
+			self.msg(gettext("[INFO]: regmark to regmark spacing Y ")+str(reg_length))
+
+		# Register Mark #
+		mm_to_user_unit = self.svg.unittouu('1mm')
+
+		# Create a new register mark layer
+		regmark_layer = self.svg.add(Layer.new(REGMARK_LAYERNAME, id=REGMARK_LAYER_ID))
```

Comment:
- currently the safearea covers anything else, so you won't see or print anything.
  - you may choose `.insert` over `.add` and choose a default value for smart layer sorting
- and the safearealayer isn't distinguishable except if you manually set `pagecheckerboard=true`
  - so you can automate and set pagecheckerboard with this plugin for a nice UX

Reply:
Yup `.insert(0, ... )` is the approach here.